### ISSUE TITLE
Weird (incorrect) ordering of contacts in groups, when allowing "show offline contacts"

### ISSUE THREAD

**Issue by Sacolevas** [emesene/emesene#994]
Hello, 

I want to report the following issue which I found strange. In my case, I like to see online and offline contacts and here is what happens:

In all messenger programs (with the sole exception of emesene2), if you allow showing offline contacts and have different groups, the way contacts are shown in each group is:
- First, online contacts are shown, ordered alphabetically.
- Then, below, offline contacts are shown, ordered alphabetically.

Here, in emesene2, if you separate groups (the only way to do that is to choose "order by group" in the menu), contacts in each groups are shown alphabetically, mixing online and offline contacts, assuming in the menu, after displaying the "Show" option,  you choose "sort by name". If you don't choose "sort by name", effectively, in each group you have (a) First, online contacts. (b) Then, offline contacts. But in both (a) and (b), the order contacts are shown is apparently random!!!!! (or at least, it is not by nickname or email, so it might be a problem here)

 If you don't separate groups (by choosing "order by status"), you see first, online contacts alphabetically, then offline contacts, but you don't see them separated by groups. 

In other words, you don't get the desired behavior in emesene2, not even by playing with the sorting options the program provides, Perhaps, the problem is in the part of "apparently random" behavior I described above (when choosing "sort by group" and unchecking "sort by name").

Thanks for your attention and this wonderful instant messaging program.

**Comment by Sacolevas** [comment]
Is there someone who can comment on this issue? Could this issue be solved? Thanks

**Comment by Sbte** [comment]
Easily I guess. But this is something very easy for starting developers, so if one of them wants to take it they can do it. Easy bugs that are not fixed before the next release are fixed by us when we go through all the issues that are left open.

Unless you want this right now; then I might look into it :P

**Comment by Knightlain** [comment]
I agree in that error, but to avoid randomness, we should specify what order we want if we don't want it by name (email for example). What order would the emesene staff prefer? Or we can use radio button for ordering. I am interested in coding it by the way.

**Comment by Sbte** [comment]
@Knightlain, what we have now, but then with preferring status first sounds ok.

**Comment by Sacolevas** [comment]
In the latest version 2.12.3 the issue has not been solved. Consequently, I don't understand why this issue is marked as closed. Thanks for your attention

**Comment by pepeleproso** [comment]
The fix is in the current dev version, you can use it from GIT. It will be available in the next stable version.

**Comment by Sacolevas** [comment]
Hello,

in the latest version 2.12.5 the fix was not applied. Actually, the same problem applies and contacts are not ordered correctly; in the option I described at the beginning, contacts seem to appear at random (I initially thought they were ordered by status, but is not the case).  I usually like to see groups, hide offline contacts and sort those contacts by name. I choose that option, but currently I see online contacts in every group but these do not appear ordered alphabetically (al least not by nickname) or any other recognizable criteria.

**Comment by c10ud** [comment]
proof please, feel free to hide mails and sensible info, we only care about initial characters

**Comment by moneycat** [comment]
@c10ud @pepeleproso Now I understand the problem and can clarify it:

My setting is only two options enabled:
- Order by group
- Sort by name

The problem of @Sacolevas may be due to: Even in each group, the contact are sorted by status first.
http://imageshack.us/photo/my-images/269/ordererror.png/

**Comment by moneycat** [comment]
On the other hand, if enable `show offline contacts`, the offline part in each group not really sorted by name.
http://imageshack.us/photo/my-images/406/ordererror2.png/
As my screenshot, the black box area is users without nickname (shown with their email). Then the whole list like 3 parts:
- users with English word first
- users with no nick but sort by email
- users with Chinese word first

The first and third part seems ordered correct (but I don't know if it is correct within third part because Chinese is more complex) because the order is English then Chinese.

**Comment by c10ud** [comment]
the second screenshot looks ok because uppercase letters have precedence over lowercase and i guess chinese characters come later in the character table.. 

the first screenshot instead seems to suggest what OP reported -- weird

**Comment by moneycat** [comment]
@c10ud You are right, the first two parts ordered correctly. I just found an user with nickname start by 'h' and its position is correctly.

**Comment by moneycat** [comment]
I just another issue. The `sort by name` seems useless in `order by status`. In my imaginary, without the sort option, it will sort by email?

**Comment by c10ud** [comment]
maybe? :p long time since i last touched that code, i'm not really sure here...contact list is always a mess to work with!

**Comment by moneycat** [comment]
@Sacolevas Please try this two new commits. Now the behaviour is more like WLM instead of using `order by status` and `sort contacts by name`.

**Comment by Sacolevas** [comment]
I have tested the latest commits (I don't know if I did it right, I just downloaded the latest emesene from github) and it finally works as desired; as @moneycat commented above, the menu options could be simplified, but at least finally you can find a combination of options to make it work.

Thanks a lot !!
